test/trento/infrastructure/software_updates/adapter/suma_http_executor_test.exs

```diff
@@ -0,0 +1,31 @@
+defmodule Trento.Infrastructure.SoftwareUpdates.Adapter.SumaHttpExecutorTest do
+  @moduledoc false
+
+  use ExUnit.Case
+
+  alias Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor
+
+  def load_certificate_content(name) do
+    File.cwd!()
+    |> Path.join("/test/fixtures/cert")
+    |> Path.join(name)
+    |> File.read!()
+  end
+
+  describe "Http executor certificate handling" do
+    test "should support chained certificates" do
+      scenarios = [
+        %{cert: "one_entry_chain.pem", expected_entries: 1},
+        %{cert: "two_entries_chain.pem", expected_entries: 2},
+        %{cert: "three_entries_chain.pem", expected_entries: 3}
+      ]
+
+      for %{cert: cert, expected_entries: expected_entries} <- scenarios do
+        assert cert
+               |> load_certificate_content()
+               |> HttpExecutor.get_cert_der()
+               |> length() == expected_entries
+      end
+    end
+  end
```

Comment:
I'm only left longing for more comprehensive testing, e.g. checking that the CA chain can actually verify a server certificate previously signed with it, but I realize this is a lot of work, and at the moment time is of the essence.

Maybe @gagandeepb could pick this up at a later stage, since he mentioned he wanted to do some testing in the context of our TLS usage in general.

Reply:
Definitely something to look into. Thanks for the pointer.

Reply:
I agree with you @stefanotorresi but at the same time this is also something on which we need a lot of feedback by the users :+1: